File: problems/split_string/splitString.php
<?php

function splitString($str, $delimiter)
{
  $chars = str_split($str);
  $words = [];
  $current = "";

  foreach($chars as $char)
  {
    if ($char === $delimiter)
    {
      $words[] = $current;
      $current = "";
    }
    else
    {
      $current .= $char;
    }
  }
  $words[] = $current;
  return $words;
}

var_dump(splitString("hello, world", ","));
var_dump(splitString("hello world", ","));
var_dump(splitString("", ","));

File: problems/split_string/split_string.rs
fn str_split<T: Into<String>>(string: T, delim: char) -> Vec<String> {
    let mut words = vec![];
    let mut curr = String::new();

    for c in string.into().chars() {
        match c == delim {
            true => {
                words.push(curr);
                curr = String::new();
            },
            false => curr.push(c),
        }
    }
    words.push(curr);
    words
}

fn main() {
    println!("{:?}", str_split("hello, world", ','));
}

